src/features/surveys/utils/getSurveyUrl.ts

```diff
@@ -1,10 +1,24 @@
 import { ZetkinSurvey } from 'utils/types/zetkin';
 
-export default function getSurveyUrl(survey: ZetkinSurvey | null) {
+export default function getSurveyUrl(
+  survey: ZetkinSurvey | null,
+  orgId: number
+) {
   if (survey) {
-    return `/organize/${survey.organization.id}/projects/${
-      survey.campaign ? `${survey.campaign.id}` : 'standalone'
-    }/surveys/${survey.id}`;
+    const campId = getSurveyCampId(survey, orgId);
+    return `/organize/${orgId}/projects/${campId}/surveys/${survey.id}`;
+  } else {
+    return '';
+  }
+}
+
+export function getSurveyCampId(survey: ZetkinSurvey | null, orgId: number) {
+  if (survey) {
+    return survey.organization.id !== orgId
+      ? 'inherited'
+      : survey.campaign
+      ? survey.campaign.id
+      : 'standalone';
```

Comment:
I struggle to read nested ternaries like this. I would recommend the following solutions instead, for readability:

```tsx
if (survey.organization.id !== orgId) {
  return 'inherited';
}

return survey.campaign?.id ?? 'standalone';
```